integrations/github-pr-review-ai-assistant/agent.py

```diff
@@ -0,0 +1,114 @@
+import os
+
+from ai_engine import UAgentResponse, UAgentResponseType
+from codedog.actors.reporters.pull_request import PullRequestReporter
+from codedog.chains import CodeReviewChain, PRSummaryChain
+from codedog.retrievers import GithubRetriever
+from github import Github
+from github.GithubException import UnknownObjectException
+from langchain.chat_models import ChatOpenAI
+from uagents import Agent, Context, Model, Protocol
+from uagents.setup import fund_agent_if_low
+
+OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
+AGENT_MAILBOX_KEY = os.getenv("AGENT_MAILBOX_KEY")
+SEED_PHRASE = os.getenv("SEED_PHRASE")
+
+agent = Agent(
+    name="Github PR review AI Assistant",
+    seed=SEED_PHRASE,
+    mailbox=f"{AGENT_MAILBOX_KEY}@https://agentverse.ai",
+)
+
+fund_agent_if_low(agent.wallet.address())
+
+
+class CodeReviewRequest(Model):
+    """Model for handling code review requests."""
+
+    repository: str
+    pull_request_number: int
+
+
+github_pr_review_protocol = Protocol("Github PR Review Automation")
+
+
+def fetch_and_review_pull_request(
+    repository: str, pull_request_number: int, context: Context
+):
+    """Fetches and reviews a GitHub pull request, returning a formatted report."""
+    if not OPENAI_API_KEY:
+        context.logger.error("OPENAI_API_KEY environment variables are not set.")
+        return "Error: Missing environment variables."
+
+    if "https://github.com/" in repository:
+        parts = repository.replace("https://github.com/", "").split("/")
+        if len(parts) >= 2:
+            repository = "/".join(parts[:2])
```

Comment:
as per [my comment](https://github.com/fetchai/uAgents/pull/262/files#r1526270825) this can be reworked using whole URL.
```suggestion
    p = urlparse('https://github.com/fetchai/uAgents/pull/262')
    if p.netloc == 'github.com':
        repository = '/'.join(p.path[1:].split('/')[:2])
```